src/core/reference/include/openvino/reference/multinomial.hpp

```diff
@@ -85,49 +86,49 @@ void multinomial(const T* probs,
     // Normalize the cdf by dividing all elements by the max value in each batch
     std::vector<T> max_value_per_batch_divisor(total_inputs_elements_count);
     ov::AxisSet target_axis_set = ov::AxisSet({last_axis});
-    broadcast(static_cast<const char*>(max_value_per_batch.data()),
-              static_cast<char*>(max_value_per_batch_divisor.data()),
+    broadcast(reinterpret_cast<const char*>(max_value_per_batch.data()),
+              reinterpret_cast<char*>(max_value_per_batch_divisor.data()),
               max_value_per_batch_shape,
-              probs_shape, // expand to original shape (expands last dim)
+              probs_shape,  // expand to original shape (expands last dim)
               target_axis_set,
               sizeof(T));
-    divide(static_cast<const T*>(cdf.data()),
-            static_cast<const T*>(max_value_per_batch_divisor.data()),
-            static_cast<T*>(cdf.data()),
-            total_inputs_elements_count,
-            false);
+    divide(reinterpret_cast<const T*>(cdf.data()),
```

Comment:
```suggestion
    divide(cdf.data(),
```
For cdf the cast shouldn't be required as `cdf` vector data type is `T`.

Reply:
Removed!